cmd/restic/exclude.go

```diff
@@ -292,3 +293,54 @@ func rejectResticCache(repo *repository.Repository) (RejectByNameFunc, error) {
 		return false
 	}, nil
 }
+
+func rejectBySize(maxSizeStr string) (RejectFunc, error) {
+	maxSize, err := parseSizeStr(maxSizeStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(item string, fi os.FileInfo) bool {
+		// directory will be ignored
+		if fi.IsDir() {
+			return false
+		}
+
+		filesize := fi.Size()
+		if filesize > maxSize {
+			debug.Log("file %s is oversize: %d", item, filesize)
+			return true
+		}
+
+		return false
+	}, nil
+}
+
+func parseSizeStr(sizeStr string) (int64, error) {
+	var (
+		numStr string
+		unit   int64
+	)
+	switch sizeStr[len(sizeStr)-1] {
+	case 'b', 'B':
+		numStr = sizeStr[:len(sizeStr)-1]
```

Comment:
Please initialize `numStr` with `numStr := sizeStr[:len(sizeStr)-1]`, to get rid of the duplicated lines. The default case can later on still set it to the full string if necessary.